graylog2-web-interface/src/views/components/MigrateFieldCharts.jsx

```diff
@@ -0,0 +1,207 @@
+// @flow strict
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { maxBy, values, isEmpty } from 'lodash';
+import Immutable from 'immutable';
+
+import { CurrentViewStateStore } from 'views/stores/CurrentViewStateStore';
+import { ViewStatesActions } from 'views/stores/ViewStatesStore';
+import SearchActions from 'views/actions/SearchActions';
+
+import Store from 'logic/local-storage/Store';
+import { widgetDefinition } from 'views/logic/Widgets';
+import AggregationWidget from 'views/logic/aggregationbuilder/AggregationWidget';
+import AggregationWidgetConfig from 'views/logic/aggregationbuilder/AggregationWidgetConfig';
+import Pivot from 'views/logic/aggregationbuilder/Pivot';
+import Series from 'views/logic/aggregationbuilder/Series';
+import WidgetPosition from 'views/logic/widgets/WidgetPosition';
+import LineVisualizationConfig from 'views/logic/aggregationbuilder/visualizations/LineVisualizationConfig';
+import AreaVisualizationConfig from 'views/logic/aggregationbuilder/visualizations/AreaVisualizationConfig';
+import type { InterpolationMode } from 'views/logic/aggregationbuilder/visualizations/Interpolation';
+
+import { Alert, Button, Row, Col } from 'components/graylog';
+import Spinner from 'components/common/Spinner';
+
+type LegacySeries = 'mean' | 'max' | 'min' | 'total' | 'count' | 'cardinality';
+type LegacyInterpolation = 'linear' | 'step-after' | 'basis' | 'bundle' | 'cardinal' | 'monotone';
+type LegacyInterval = 'minute' | 'hour' | 'day' | 'week' | 'month' | 'quarter' | 'year'
+type LegacyVisualization = 'bar' | 'area' | 'line' | 'scatterplot'
+type LegacyFieldChart = {
+  field: string,
+  renderer: LegacyVisualization,
+  interpolation: LegacyInterpolation,
+  valuetype: LegacySeries,
+  interval: LegacyInterval,
+}
+
+const Actions = styled.div`
+  margin-top: 10px;
+`;
+
+const mapSeries = (legacySeries: LegacySeries, field: string) => {
+  let seriesName;
+  switch (legacySeries) {
```

Comment:
I would suggest extracting the `switch`-statement into a separate `mapFunction` fn. The benefit is that you can do a simple `return` in each `case`-branch and do not have to assign a variable, making the whole construct easier to understand. E.g. for `total`:
```
  case 'total':
    return 'sum';
```

`mapSeries` would then just look like this: `(legacySeries, field) => `${mapFunction(legacySeries)}(${field});`